apps/src/block_utils.js

```diff
@@ -545,14 +546,21 @@ const LABELED_INPUT_PARTS_REGEX = /(.*?)({([^}]*)}|\n|$)/m;
  * Finds the input config for the given input name, and removes it from args.
  * @param {InputConfig[]} args List of configs to search through
  * @param {string} inputName name of input to find and remove
+ * @param {string} blockText original block text used for metrics reporting in the case of a missing input
  * @returns InputConfig the input config with name `inputName`
  */
-const findAndRemoveInputConfig = (args, inputName) => {
-  const argIndex = args.findIndex(arg => arg.name === inputName);
-  if (argIndex !== -1) {
-    return args.splice(argIndex, 1)[0];
+const findAndRemoveInputConfig = (args, inputName, blockText) => {
+  let argIndex = args.findIndex(arg => arg.name === inputName);
+  if (argIndex === -1) {
+    // In the case of a missing input, default to the first available arg.
+    argIndex = 0;
+    MetricsReporter.logWarning({
+      event: 'BLOCK_MISSING_INPUT',
+      message: `${inputName} not found in args. Defaulting to ${args[argIndex].name}`,
+      blockText,
+    });
   }
-  throw new Error(`${inputName} not found in args`);
+  return args.splice(argIndex, 1)[0];
```

Comment:
I didn't realize that `splice` returned an array with the removed elements!
This is not your code and I don't have a strong preference, but what do you think of this change to make it more clear to readers? 
```
const removedInputArray = args.splice(argIndex, 1);
return removedInputArray[0];
```

Reply:
Hmm, I think I personally do like the efficiency of keeping it in one line, since it's basically equivalent to "remove and return the element at this position"?